File: scripts/generate_deck.py
from pptx import Presentation
from pptx.util import Inches, Pt
from pptx.dml.color import RGBColor
from pptx.enum.text import PP_ALIGN

# Colors (RGB)
BG_COLOR = RGBColor(0x1A, 0x0B, 0x2E) # Deep Purple
ACCENT_COLOR = RGBColor(0xF5, 0x9E, 0x0B) # Gold/Amber
TEXT_MAIN = RGBColor(0xFF, 0xFF, 0xFF)
TEXT_SUB = RGBColor(0xE9, 0xD5, 0xFF) # Light Purple

def create_deck():
    prs = Presentation()
    
    # helper to set background
    def set_bg(slide):
        background = slide.background
        fill = background.fill
        fill.solid()
        fill.fore_color.rgb = BG_COLOR

    # 1. Title Slide
    slide_layout = prs.slide_layouts[0] # Title Slide
    slide = prs.slides.add_slide(slide_layout)
    set_bg(slide)
    
    title = slide.shapes.title
    subtitle = slide.placeholders[1]
    
    title.text = "SajuChain"
    title.text_frame.paragraphs[0].font.color.rgb = ACCENT_COLOR
    title.text_frame.paragraphs[0].font.size = Pt(60)
    title.text_frame.paragraphs[0].font.bold = True
    
    subtitle.text = "Destiny on Chain: Where Ancient Wisdom Meets Eternal Ledger\nInvestor Presentation 2026"
    subtitle.text_frame.paragraphs[0].font.color.rgb = TEXT_MAIN
    subtitle.text_frame.paragraphs[0].font.size = Pt(24)

    # Helper for Content Slides
    def add_content_slide(title_text, content_text):
        slide_layout = prs.slide_layouts[1] # Title and Content
        slide = prs.slides.add_slide(slide_layout)
        set_bg(slide)
        
        title = slide.shapes.title
        title.text = title_text
        title.text_frame.paragraphs[0].font.color.rgb = ACCENT_COLOR
        title.text_frame.paragraphs[0].font.bold = True
        
        body = slide.placeholders[1]
        tf = body.text_frame
        tf.text = content_text
        
        for p in tf.paragraphs:
            p.font.color.rgb = TEXT_MAIN
            p.font.size = Pt(18)
            
    # 2. Problem
    add_content_slide(
        "The Problem: Uncertainty & Ephemerality", 
        "• Subjective Interpretation: Relies on individual skill, inconsistent results.\n"
        "• Lack of Permanence: Physical amulets lost. No digital record.\n"
        "• Data Privacy: Sensitive birth data shared with unverified services."
    )

    # 3. Solution
    add_content_slide(
        "The Solution: SajuChain",
        "1. Algorithmic Precision: lunar-javascript + GPT-4o.\n"
        "2. Immutable Ownership: Minted as Compressed NFTs (cNFT) on Solana.\n"
        "3. Digital Mysticism: Ritualistic UI/UX."
    )

    # 4. Tech
    add_content_slide(
        "Technical Architecture",
        "• Frontend: Next.js 16, Tailwind CSS v4, Framer Motion\n"
        "• Blockchain: Solana Web3.js, Metaplex (Umi)\n"
        "• AI Engine: Hybrid Logic (Deterministic + Generative)"
    )

    # 5. Roadmap
    add_content_slide(
        "Roadmap & Vision",
        "• Phase 1 (Now): MVP, Devnet, UI/UX V2 (Completed)\n"
        "• Phase 2 (Q3 2026): Mainnet, Mobile App\n"
        "• Phase 3 (2027): Oracle DAO, Metaverse Identity"
    )

    prs.save('sajuchain_pitch_deck.pptx')
    print("Created sajuchain_pitch_deck.pptx")

if __name__ == "__main__":
    create_deck()

File: scripts/generate_deck.js
const PptxGenJS = require("pptxgenjs");

const pres = new PptxGenJS();

// Theme Colors
const COLORS = {
    BG: "1A0B2E", // Deep Purple
    ACCENT: "F59E0B", // Gold/Amber
    TEXT_MAIN: "FFFFFF",
    TEXT_SUB: "E9D5FF", // Light Purple
};

// Layout
pres.layout = "LAYOUT_16x9";
pres.background = { color: COLORS.BG };

// Helper to add slide with common styling
function addSlide(title) {
    const slide = pres.addSlide();
    slide.background = { color: COLORS.BG };

    // Add "Confidential" footer
    slide.addText("SajuChain - Confidential", {
        x: 0.5, y: 7.0, w: "90%", h: 0.3,
        fontSize: 10, color: "555555", align: "center"
    });

    if (title) {
        slide.addText(title, {
            x: 0.5, y: 0.5, w: "90%", h: 1.0,
            fontSize: 32, fontFace: "Georgia", color: COLORS.ACCENT, bold: true,
            align: "left"
        });
        // Decorative line
        slide.addShape(pres.ShapeType.line, {
            x: 0.5, y: 1.4, w: "90%", h: 0,
            line: { color: COLORS.ACCENT, width: 2 }
        });
    }
    return slide;
}

// 1. Title Slide
const slide1 = pres.addSlide();
slide1.background = { color: COLORS.BG };
slide1.addText("SajuChain", {
    x: 0, y: 2.5, w: "100%", h: 1.5,
    fontSize: 60, fontFace: "Georgia", color: COLORS.ACCENT, bold: true, align: "center"
});
slide1.addText("Destiny on Chain: Where Ancient Wisdom Meets Eternal Ledger", {
    x: 1, y: 4.0, w: "80%", h: 1,
    fontSize: 24, fontFace: "Arial", color: COLORS.TEXT_MAIN, align: "center"
});
slide1.addText("Investor Presentation 2026", {
    x: 0, y: 6.5, w: "100%", h: 0.5,
    fontSize: 14, color: COLORS.TEXT_SUB, align: "center"
});

// 2. The Problem
const slide2 = addSlide("The Problem: Uncertainty & Ephemerality");
slide2.addText([
    { text: "Subjective Interpretation\n", options: { fontSize: 24, bold: true, breakLine: true } },
    { text: "Fortune telling relies on individual skill, leading to inconsistent results.\n\n", options: { fontSize: 18 } },
    { text: "Lack of Permanence\n", options: { fontSize: 24, bold: true, breakLine: true } },
    { text: "Physical amulets are lost. No permanent digital record of one's destiny.\n\n", options: { fontSize: 18 } },
    { text: "Data Privacy\n", options: { fontSize: 24, bold: true, breakLine: true } },
    { text: "Sensitive birth data shared with unverified services.", options: { fontSize: 18 } }
], { x: 1.0, y: 2.0, w: "80%", h: 4.5, color: COLORS.TEXT_MAIN });

// 3. The Solution
const slide3 = addSlide("The Solution: SajuChain");
slide3.addText("A Tripartite Architecture", { x: 0.5, y: 1.6, w: "90%", h: 0.5, fontSize: 14, color: COLORS.TEXT_SUB, italic: true });

// Column 1
slide3.addText("Algorithmic Precision", { x: 0.5, y: 2.5, w: 4, h: 0.5, fontSize: 20, color: COLORS.ACCENT, bold: true });
slide3.addText("lunar-javascript + GPT-4o for nuanced, accurate readings.", { x: 0.5, y: 3.0, w: 3.5, h: 2, fontSize: 14, color: COLORS.TEXT_MAIN });

// Column 2
slide3.addText("Immutable Ownership", { x: 4.5, y: 2.5, w: 4, h: 0.5, fontSize: 20, color: COLORS.ACCENT, bold: true });
slide3.addText("Minted as Compressed NFTs (cNFT) on Solana. True 'Soulbound' ownership.", { x: 4.5, y: 3.0, w: 3.5, h: 2, fontSize: 14, color: COLORS.TEXT_MAIN });

// Column 3
slide3.addText("Digital Mysticism", { x: 8.5, y: 2.5, w: 4, h: 0.5, fontSize: 20, color: COLORS.ACCENT, bold: true });
slide3.addText("A ritualistic UI experience designed to evoke reverence.", { x: 8.5, y: 3.0, w: 3.5, h: 2, fontSize: 14, color: COLORS.TEXT_MAIN });

// 4. Technology
const slide4 = addSlide("Technical Architecture");
slide4.addText([
    { text: "Frontend: ", options: { bold: true, color: COLORS.ACCENT } },
    { text: "Next.js 16 (App Router), Tailwind CSS v4, Framer Motion\n\n" },
    { text: "Blockchain: ", options: { bold: true, color: COLORS.ACCENT } },
    { text: "Solana Web3.js, Metaplex (Umi), Compressed NFTs\n\n" },
    { text: "AI Engine: ", options: { bold: true, color: COLORS.ACCENT } },
    { text: "Hybrid Logic (Deterministic Saju + Generative LLM)" }
], { x: 1.0, y: 2.0, w: "80%", h: 4.0, fontSize: 18, color: COLORS.TEXT_MAIN, bullet: true });

// 5. Future Vision
const slide5 = addSlide("Future Vision: The Oracle DAO");
slide5.addText([
    { text: "Phase 1: Genesis (Now)", options: { fontSize: 20, bold: true, color: COLORS.ACCENT } },
    { text: "MVP, Devnet, UI/UX Overhaul\n\n", options: { fontSize: 16 } },

    { text: "Phase 2: Expansion (Q3 2026)", options: { fontSize: 20, bold: true, color: COLORS.ACCENT } },
    { text: "Mainnet, Mobile App, Social Compatibility\n\n", options: { fontSize: 16 } },

    { text: "Phase 3: Oracle DAO (2027)", options: { fontSize: 20, bold: true, color: COLORS.ACCENT } },
    { text: "Decentralized interpretation market, Metaverse ID", options: { fontSize: 16 } }
], { x: 1.0, y: 2.0, w: "80%", h: 4.5, color: COLORS.TEXT_MAIN });

// Save
pres.writeFile({ fileName: "sajuchain_pitch_deck.pptx" })
    .then(fileName => {
        console.log(`Created file: ${fileName}`);
    })
    .catch(err => {
        console.error(err);
    });
